pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -447,9 +496,10 @@ def _result_id_type(self) -> _ResultIdTuple:
         return (str, int, int, str)
 
     def rebase_pass(self) -> BasePass:
-        return auto_rebase_pass(
-            {OpType.CX, OpType.X, OpType.SX, OpType.Rz},
-        )
+        if self._primitive_gates == GateSet.X_SX_RZ_CX.value:
+            return auto_rebase_pass(GateSet.X_SX_RZ_CX.value)
+        elif self._primitive_gates == GateSet.X_SX_RZ_ECR.value:
+            return ecr_rebase
```

Comment:
Why not return `ecr_rebase` if `self._primitive_gates` allows it and otherwise return `auto_rebase_pass(self._primitive_gates)`? That will then raise an exception if we get a gateset that we can't rebase to.

Reply:
Done, this is an improvement

Reply:
Do you think raising an excpetion in the case of an unsupported gateset is sufficent? This exception would be given when the user does the default compilation pass.

I got the impression that you also wanted an exception rasied when the user tried to initialise an `IBMQBackend` object but maybe I misunderstood.

Reply:
> Do you think raising an excpetion in the case of an unsupported gateset is sufficent here? This exception would be given when the user does the default compilation pass.
> 
> I got the impression that you also wanted an exception rasied when the user tried to construct an `IBMQBackend` object with an unsupported gateset but maybe I misunderstood.

We can constuct the rebase pass on initialization (and then store it as an attribute).

Reply:
Store the rebase pass as an attribute? I'm not sure I follow. 

Currently the primitive gates are an attribute which is used to compute the rebase in the `rebase_pass` method.

 Should I store the output of `rebase_pass` as an attribute?

Reply:
That's what I'm suggesting if you want to fail on backend construction but actually I've changed my mind, I don't think this is a good idea: better to fail on compilation. That allows users to submit circuits that satisfy the gateset even if the default compilation passes don't support it.

Reply:
Yeah I agree I think its better to fail on compilation as well in this case. 

Thanks